### ISSUE TITLE
Compilation error with gcc 9.3.0 on Ubuntu 20.04

### ISSUE THREAD

**Issue by quicky2000** [oneapi-src/oneTBB#312]
After cloing the repository and performing cmake <repository root>
cmake --build <build_dir> failed with the following error:

```
[ 40%] Building CXX object test/CMakeFiles/test_concurrent_set.dir/tbb/test_concurrent_set.cpp.o
In file included from /media/data/Programmation/C++/build/oneTBB/oneTBB/test/common/concurrent_ordered_common.h:20,
                 from /media/data/Programmation/C++/build/oneTBB/oneTBB/test/tbb/test_concurrent_set.cpp:21:
/media/data/Programmation/C++/build/oneTBB/oneTBB/test/common/concurrent_associative_common.h: In instantiation of ‘void test_scoped_allocator() [with ContainerTraits = COSetTraits]’:
/media/data/Programmation/C++/build/oneTBB/oneTBB/test/tbb/test_concurrent_set.cpp:198:40:   required from here
/media/data/Programmation/C++/build/oneTBB/oneTBB/test/common/concurrent_associative_common.h:1184:8: error: implicitly-declared ‘tbb::detail::d1::concurrent_set<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > >, std::less<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > >, std::scoped_allocator_adaptor<std::allocator<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > > > >& tbb::detail::d1::concurrent_set<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > >, std::less<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > >, std::scoped_allocator_adaptor<std::allocator<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > > > >::operator=(const tbb::detail::d1::concurrent_set<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > >, std::less<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > >, std::scoped_allocator_adaptor<std::allocator<AllocatorAwareData<std::scoped_allocator_adaptor<std::allocator<int> > > > > >&)’ is deprecated [-Werror=deprecated-copy]
 1184 |     c1 = c2;
      |     ~~~^~~~
compilation terminated due to -Wfatal-errors.
```

**Comment by johhnry** [comment]
Hi,

I can confirm this bug on Pop!_OS 20.10 with `gcc (Ubuntu 10.2.0-13ubuntu1) 10.2.0` by running an out of source build.

**Comment by alexey-katranov** [comment]
@kboyarinov, can you please follow up on the issue?

**Comment by johhnry** [comment]
Hi, I just compiled successfully on Pop!_OS 20.10 with the last version d86ed7f ! Thanks guys

**Comment by kboyarinov** [comment]
@quicky2000, @johhnry,

This issue was fixed in [oneTBB 2021.2.0](https://github.com/oneapi-src/oneTBB/releases/tag/v2021.2.0).
If you still have any issues, feel free to contact us.
